cvat/apps/engine/frame_provider.py

```diff
@@ -429,8 +437,30 @@ def _get_segment(self, validated_frame_number: int) -> models.Segment:
 
         return segment
 
+    def unload(self):
+        self._clear_segment_frame_provider_cache()
+
+    def _clear_segment_frame_provider_cache(self):
+        while self._segment_frame_provider_cache:
+            cached_provider = self._segment_frame_provider_cache.popitem()[1]
+            if sys.getrefcount(cached_provider) == 2:
```

Comment:
This doesn't seem very robust. It relies on specifics of the reference counting implementation, so it's liable to silently change behavior when moving between Python versions. The docs even say:

> Consequently, do not rely on the returned value to be accurate, other than a value of 0 or 1.

Moreover, is it even needed? As far as I can see, every time `_get_segment_frame_provider` is called, the resulting provider is used and discarded immediately. So it shouldn't be possible for the cached provider to be used anywhere at this point.

Reply:
Not really, in the cases known to me it's going to be cleaned by the GC after the `cache.clear()` call, once there are no references.